xmodule/x_module.py

```diff
@@ -1536,11 +1535,8 @@ def service(self, block, service_name):
         Returns:
             An object implementing the requested service, or None.
         """
-        declaration = block.service_declaration(service_name)
-        service = self._runtime_services.get(service_name)
-        if declaration is None or service is None:
-            # getting the service from parent module. making sure of block service declarations.
-            service = super().service(block=block, service_name=service_name)
+        # Getting the service from parent module. making sure of block service declarations.
+        service = super().service(block=block, service_name=service_name)
```

Comment:
We don't need `runtime._runtime_services` anymore. Within a single request, each XBlock uses the same instance of the runtime, so we can safely rely on the `XBlock` library to check the service declaration:
https://github.com/openedx/XBlock/blob/7b8bdeafbabbbc96f11dfbceaf53c89bd416ea6f/xblock/runtime.py#L1120-L1126.